internal/api/core/server.go

```diff
@@ -65,25 +66,68 @@ func NewPersesAPI(serviceManager dependency.ServiceManager, persistenceManager d
 		configendpoint.New(cfg),
 		migrateendpoint.New(serviceManager.GetMigration()),
 		validateendpoint.New(serviceManager.GetSchemas(), serviceManager.GetDashboard()),
-		authendpoint.New(persistenceManager.GetUser(), serviceManager.GetCrypto()),
+		authendpoint.New(persistenceManager.GetUser(), serviceManager.GetJWT()),
 	}
 	return &api{
 		apiV1Endpoints: apiV1Endpoints,
 		apiEndpoints:   apiEndpoints,
+		jwtMiddleware: serviceManager.GetJWT().Middleware(func(c echo.Context) bool {
+			return !*cfg.ActivatePermission
+		}),
 	}
 }
 
 func (a *api) RegisterRoute(e *echo.Echo) {
-	a.registerAPIV1Route(e)
+	// First, let's collect every route.
+	// The expecting result is a tree we will need to loop over.
+	groups := a.collectRoutes()
+	// Now let's create a simple struct that will help us to loop over the route tree.
+	type queue struct {
```

Comment:
When I saw this I thought about queue data struct.
But I saw on the comment route tree I think it might be a bit confuse.
Just sharing a felling, I don't have a better name suggestion yet.

Reply:
yeah I agree the name is not accurate since it's not the queue but a element of the queue actually.
Maybe `queueNode` ?

Reply:
yeah, `queueNode` seems better, another suggestion would be `queueElement` feel free to chose anyone, I don't have preference over those.